clang-tools-extra/test/clang-tidy/checkers/modernize/use-digit-separator.cpp

```diff
@@ -0,0 +1,561 @@
+// RUN: %check_clang_tidy -std=c++14-or-later %s modernize-use-digit-separator %t
+
+
+// Long not formatted literals
+
+int NotFormattedInteger = 1234567;
+// CHECK-MESSAGES: :[[@LINE-1]]:27: warning: unformatted representation of integer literal '1234567' [modernize-use-digit-separator]
+// CHECK-FIXES: 1'234'567
```

Comment:
Please add a test case like this:
```
int OldStyleFormattedInteger = 12*1000*1000;
// CHECK-MESSAGES: :[[@LINE-1]]:27: warning: legacy formatted representation of integer literal '12*1000*1000' [modernize-use-digit-separator]
// CHECK-FIXES: 12'000'000
```

See https://github.com/llvm/llvm-project/issues/35414#issuecomment-2880248198 for more details

Reply:
I disagree with this suggestion; see my comments on the original issue.  A check should have a single responsibility, not two.